src/content_script/mastodonInject.js

```diff
@@ -0,0 +1,143 @@
+    window.open(`/users/${username}/remote_follow`, "_blank");
+}
+
+/**
+ * Replacement onClick handler for interaction buttons.
+ * 
+ * @param {Event} event
+ * @returns {void}
+ */
+function onClickInteract(event) {
+    event.stopPropagation();
+    event.preventDefault();
+    const articleElement = event.target.closest("article[data-id]");
+    const tootId = (
+        articleElement === null
+            ? window.location.pathname.split("/").slice(-1)[0]
+            : articleElement.getAttribute("data-id")
+    );
+    // activate AutoRemoteFollow
+    window.open(`/interact/${tootId}`, "_blank");
+}
+
+/**
+ * Wait for element to appear.
+ * 
+ * @param {string} selector
+ * @param {boolean} multiple
+ * @param {number} timeoutDuration
+ * @see {@link https://github.com/storybookjs/test-runner/blob/6d41927154e8dd1e4c9e7493122e24e2739a7a0f/src/setup-page.ts#L134}
+ *  from which this was adapted
+ * @returns {Promise}
+ */
+function waitForElement(selector, multiple = false, timeoutDuration) {
+    return new Promise((resolve, reject) => {
+        const getElement = () => (
+            multiple
+                ? document.querySelectorAll(selector)
+                : document.querySelector(selector)
+        );
+        const isElementFound = (el) => (!multiple && el) || (multiple && el.length > 0);
+
+        const timeout = window.setTimeout(() => {
+            reject(new Error("waitForElement timed out"));
+        }, timeoutDuration);
+
+        const element = getElement();
+        if(isElementFound(element)){
+            window.clearTimeout(timeout);
+            return resolve(element);
+        }
+
+        const observer = new MutationObserver(() => {
+            const element = getElement();
+            if(isElementFound(element)){
+                window.clearTimeout(timeout);
+                resolve(element);
+                observer.disconnect();
+            }
+        });
+
+        observer.observe(document.body, {
+            childList: true,
+            subtree: true
+        });
+
+        return null;
+    }); 
+}
+
+/**
+ * Inject replacement onClick handler for Follow button.
+ * 
+ * @returns {void}
+ */
+async function injectFollowButton() {
+    try {
+        const followButton = await waitForElement(".account__header__tabs__buttons button:first-of-type", false, TIMEOUT_DURATION);
+        followButton.addEventListener("click", onClickFollow);
+    } catch (error) {
+        // Follow button failed to appear
+    }
+}
+
+/**
+ * Inject replacement onClick handler for Interaction buttons.
+ * 
+ * @returns {void}
+ */
+async function injectInteractionButtons() {
+    const INJECTED_REPLY_CLASS = "mastodon-simplified-federation-injected-interaction";
+    try {
+        const replyButtons = await waitForElement(
+            ".item-list[role='feed'] article[data-id] .status__action-bar button," +
+            ".detailed-status__wrapper .detailed-status__action-bar button",
```

Comment:
Did not look at the CSS now but that looks very much specific. Note again I want to avoid breaking it in v4.0.1 or so and having us to fix it again.

But I soon fiddle around with that.

Reply:
If it is not really possible, we may need to think about providing/targeting the same element with multiple selectors, then potentially combining and deduplicating them and then finally

Js-xoding wise maybe an (async) generator generating htmlelements after they are found could be a sleek way to implement that: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AsyncGenerator

But that is just a first idea/shot. As always KISS, i.e. keep it simple if you can. So if we don't need the added complexity, then leave it.
Only if it is worthy and may make the extension more stable/robust against future changes let's do it (in some esy, that's just an idea).

Reply:
I did try to make it a more generic selector by referring to the more general CSS class and attributes which are less likely to change than class names. I like the idea of targeting the same element with multiple selectors and deduplicating, but it's difficult to imagine the various ways in which the layout or the classes could change at the moment. I'm inclined not to over-engineer it for now.

Reply:
Good, I'm fine with that.